components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -3104,7 +3113,90 @@ private String saveAPI(API api,
         return success;
     }
 
+    /**
+     * This method check whether API older version exist or not
+     *
+     * @param apiObj    json object
+     * @return apiOlderVersionExist
+     * @throws APIManagementException
+     */
+    private boolean isAPIOlderVersionExist(NativeObject apiObj) throws APIManagementException {
+        boolean apiOlderVersionExist = false;
+        if (apiObj == null) {
+            handleException("Invalid number of input parameters.");
+        }
+        String provider = (String) apiObj.get("provider", apiObj);
+        provider = APIUtil.replaceEmailDomain(provider);
+        String name = (String) apiObj.get("name", apiObj);
+        String currentVersion = (String) apiObj.get("version", apiObj);
+        boolean isTenantFlowStarted = false;
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(provider));
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            }
+            Set<String> versions = getAPIVersions(provider, name);
+            APIVersionStringComparator comparator = new APIVersionStringComparator();
+            for (String version : versions) {
+                if (comparator.compare(version, currentVersion) < 0) {
+                    apiOlderVersionExist = true;
+                    break;
+                }
+            }
+        } finally {
+            if (isTenantFlowStarted) {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+        return apiOlderVersionExist;
+    }
+
+    /**
+     * This method check whether URL valid or not
+     *
+     * @param type
```

Comment:
Add descriptions for the parameters. Mention that when could the exception is thrown.